macro-ipld-derive/src/lib.rs

```diff
@@ -0,0 +1,35 @@
+use proc_macro::TokenStream;
+use quote::quote;
+use syn::{parse_macro_input, DeriveInput};
+
+#[proc_macro_derive(SerdeIpld)]
+pub fn derive_serde_ipld(input: TokenStream) -> TokenStream {
+    let ast = parse_macro_input!(input as DeriveInput);
+
+    let name = &ast.ident;
+
+    let expanded = quote! {
+            // impl MacroTrait for #name // should we make a trait?
+            impl #name {
+                /// serde serialize to dag-json
+                pub fn to_json(&self) -> Result<String, Box<dyn std::error::Error>> {
+                    let json_bytes = serde_ipld_dagjson::to_vec(self)?;
+                    String::from_utf8(json_bytes).map_err(Into::into)
+                }
+
+                /// serde serialize to dag-cbor
+                pub fn to_cbor(&self) -> Result<Vec<u8>, Box<dyn std::error::Error>> {
```

Comment:
I'd prefer not to loose the information about which concrete serde error was produced. Can we make these method return serde errors since we are calling serde methods?